cmd/crc/cmd/oc_env.go

```diff
@@ -22,7 +23,16 @@ var ocEnvCmd = &cobra.Command{
 			errors.ExitWithMessage(1, "Error running the oc-env command: %s", err.Error())
 		}
 
+		clusterConfig, err := machine.GetClusterConfig(constants.DefaultName)
```

Comment:
@cfergeau I am trying to understand why we need to create this function only to get proxy info? shouldn't we abstracted the proxy to use it for any method/function? Since now the root.cmd `prerun` happen so we know if the proxy is set of not and can be get using `network.NewProxyConfig()` . Are you trying to only show those proxy detail if the VM is running ?

Reply:
In order to be able to add `BaseDomain` to `NO_PROXY`, we need to read it from a `crc-bundle-info.json` file. At the moment, this means going through `~/.crc/machines/crc/config.json` to get the `BundleName`, and then looking up the correct bundle from `~/.crc/cache`. This is why this is not using `network.NewProxyConfig` directly.

Reply:
@cfergeau Got it, but I still think that we should get the basedomain instead of the whole cluster config.

```golang
	proxy, err := network.NewProxyConfig()
	if err != nil {
		return nil, err
	}
	if proxy.IsEnabled() {
               baseDomain := machine.GetbaseDomain()
               proxy.AddNoProxy(fmt.Sprintf(".%s", baseDomainName))

	       output.Outln(shell.GetEnvString(userShell, "HTTP_PROXY", clusterConfig.ProxyConfig.HttpProxy))
	       output.Outln(shell.GetEnvString(userShell, "HTTPS_PROXY", clusterConfig.ProxyConfig.HttpsProxy))
	       output.Outln(shell.GetEnvString(userShell, "NO_PROXY", clusterConfig.ProxyConfig.GetNoProxyString()))
	}
```

Reply:
Ah, `GetClusterConfig` dates from when I hoped to share this code with `GetConsoleURL`, this can indeed be reworked a bit

Reply:
will this rework be done now or postponed?

Reply:
I changed machine.GetClusterConfig() to machine.GetProxyConfig(). I don't want to duplicate in cmd/ the logic to create the proxy config (in particular the `proxy.AddNoProxy(fmt.Sprintf(".%s", baseDomainName))` part)